draft-irtf-cfrg-vdaf.md

```diff
@@ -4035,16 +4046,31 @@ differential privacy.
 
 The objects we describe in {{prg}} share a common interface, which we have
 called Prg. However, these are not necessarily all modeled as cryptographic
-Pseudorandom Generators in the security analyses of our protocols. Instead, most
-of them are modeled as random oracles. For these use cases, we want to be
-conservative in our assumptions, and hence prescribe PrgSha3 as the only
-RECOMMENDED Prg instantiation.
+Pseudorandom Generators (PRGs) in the security analyses of our protocols.
+Instead, most of them are modeled as random oracles. For these use cases, we
+want to be conservative in our assumptions, and hence prescribe PrgSha3 as the
+only RECOMMENDED Prg instantiation.
 
 The one exception is the PRG used in the Idpf implementation IdpfPoplar
-{{idpf-poplar}}. Here, a random oracle is not needed to prove security, and
-hence a construction based on fixed-key AES {{prg-fixed-key-aes128}} can be
-used. However, as PrgFixedKeyAes128 has been shown to be differentiable from
-a random oracle {{GKWWY20}}, it is NOT RECOMMENDED to use it anywhere else.
+{{idpf-poplar}}. Here, a random oracle is not needed to prove security, since
+the security analysis of {{BBCGGI21}}, Proposition 1, only requires a PRG.
```

Comment:
What about extractability? We used the ROM to prove our IDPF variant extractable in [[DPRS23](https://eprint.iacr.org/2023/130), Lemma 1]. Could we have maybe used correlation robustness instead?

Reply:
Hmm yeah, extractability may in fact be an issue. BBCGGI21 also assume a random oracle for the extractable DPF, although they implement it using fixed-key AES. Now the question is how bad is it if we don't have extractability. One-hotness of the *evaluated* part of the client's vector is ensured through the sketch in BBCGGI21 (or through the technique from [DP22](https://eprint.iacr.org/2021/580.pdf) in the case of Doplar). So the main concern here is the attack described in [[BBCGGI21](https://eprint.iacr.org/2021/017.pdf), Section 4.2]: A client using the fact that fixed-key AES is *not* a random oracle in order to craft a malicious DPF key pair that evaluates to 1 in more than one position, but where the servers only end up evaluating one of these positions. Personally I would feel fine with documenting this, and offering PrgSha3 as an alternative for users who want the additional robustness guarantees that extractability gives them. For privacy both fixed-key AES and SHA-3 should be fine (under the respective assumptions).

Reply:
That sounds reasonable, but:
1. We'll need to define a separate codepoint for Poplar1 with SHA-3
2. We'll need to provide concrete guidance (i.e., if attack X is possible, then users MUST choose Poplar1 with SHA-3)

Question: Is there anything about the attack you describe that's not captured by robustness as defined in DPRS23? Do you think that extractability would be needed to prove that Poplar1 is robust?

For what it's worth, I'm not sure the analysis of Doplar will be entirely analogous. We use the fact that we can extract from the inputs from the random oracle in order to then do a reduction to DFLP. This initial simplifying step necessary because if we don't know if the IDPF output is onehot, then a DFLP for the sum of the output vector is vacuous. (When we were working on the proof, we couldn't figure out a different assumption about IDPF that would have admitted the same argument.)

Reply:
I added a paragraph talking about extractability and what it means for robustness. Given that the (theoretical) attacks by malicious clients are very limited, I don't think it's necessary to define Poplar1 with SHA-3, and I doubt anyone would actually use it in practice.